src/node_jsrtapi.cc

```diff
@@ -191,380 +269,389 @@ _Ret_maybenull_ JsValueRef CALLBACK CallbackWrapper(JsValueRef callee, bool isCo
+  JsValueRef description = nullptr;
+  if (s != nullptr) {
+    CHECK_JSRT(JsCreateStringUtf8((uint8_t*)s, strlen(s), &description));
+  }
+  CHECK_JSRT(JsCreateSymbol(description, reinterpret_cast<JsValueRef*>(result)));
+  return napi_ok;
+}
+
+napi_status napi_create_error(napi_env, napi_value msg, napi_value* result) {
+  CHECK_ARG(result);
   JsValueRef message = reinterpret_cast<JsValueRef>(msg);
-  JsValueRef errorMsg = nullptr;
-  error = JsCreateError(message, &errorMsg);
-  return reinterpret_cast<napi_value>(errorMsg);
+  CHECK_JSRT(JsCreateError(message, reinterpret_cast<JsValueRef*>(result)));
+  return napi_ok;
 }
 
-napi_value napi_create_type_error(napi_env, napi_value msg) {
-  JsErrorCode error = JsNoError;
+napi_status napi_create_type_error(napi_env, napi_value msg, napi_value* result) {
+  CHECK_ARG(result);
   JsValueRef message = reinterpret_cast<JsValueRef>(msg);
-  JsValueRef errorMsg = nullptr;
-  error = JsCreateTypeError(message, &errorMsg);
-  return reinterpret_cast<napi_value>(errorMsg);
+  CHECK_JSRT(JsCreateTypeError(message, reinterpret_cast<JsValueRef*>(result)));
+  return napi_ok;
 }
 
-napi_valuetype napi_get_type_of_value(napi_env e, napi_value vv) {
-  JsErrorCode error = JsNoError;
+napi_status napi_get_type_of_value(napi_env e, napi_value vv, napi_valuetype* result) {
+  CHECK_ARG(result);
   JsValueRef value = reinterpret_cast<JsValueRef>(vv);
   JsValueType valueType = JsUndefined;
-  error = JsGetValueType(value, &valueType);
-
-  if (valueType == JsNumber) {
-    return napi_number;
+  CHECK_JSRT(JsGetValueType(value, &valueType));
+
+  switch (valueType) {
+    case JsUndefined: *result = napi_undefined; break;
+    case JsNull: *result = napi_null; break;
+    case JsNumber: *result = napi_number; break;
+    case JsString: *result = napi_string; break;
+    case JsBoolean: *result = napi_boolean; break;
+    case JsFunction: *result = napi_function; break;
+    case JsSymbol: *result = napi_symbol; break;
+    default: *result = napi_object; break;
   }
-  else if (valueType == JsString) {
-    return napi_string;
-  }
-  else if (valueType == JsFunction) {
-    // This test has to come before IsObject because IsFunction
-    // implies IsObject
-    return napi_function;
-  }
-  else if (valueType == JsObject) {
-    return napi_object;
-  }
-  else if (valueType == JsBoolean) {
-    return napi_boolean;
-  }
-  else if (valueType == JsUndefined) {
-    return napi_undefined;
-  } /*else if (v->IsSymbol()) {
-    return napi_symbol;
-    }*/ else if (valueType == JsNull) {
-    return napi_null;
-  }
-    else {
-      return napi_object;   // Is this correct?
-    }
+  return napi_ok;
 }
 
-napi_value napi_get_undefined_(napi_env e) {
-  JsValueRef undefinedValue = nullptr;
-  JsErrorCode error = JsNoError;
-  error = JsGetUndefinedValue(&undefinedValue);
-  return reinterpret_cast<napi_value>(undefinedValue);
+napi_status napi_get_undefined_(napi_env e, napi_value* result) {
```

Comment:
This API has been renamed without the underscore in V8 version (napi_get_undefined). Should match here.